core/genesis.go

```diff
@@ -282,8 +282,9 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, g
 			if overrides != nil && overrides.OverrideOptimismCanyon != nil {
 				config.CanyonTime = overrides.OverrideOptimismCanyon
 				config.ShanghaiTime = overrides.OverrideOptimismCanyon
-				if config.Optimism != nil && config.Optimism.EIP1559DenominatorCanyon == 0 {
-					config.Optimism.EIP1559DenominatorCanyon = 250
+				if config.Optimism != nil {
+					eip1559DenominatorCanyon := uint64(250)
+					config.Optimism.EIP1559DenominatorCanyon = &eip1559DenominatorCanyon
```

Comment:
Isn't this changing the old logic? The old logic said, only set the default value of 250 if the current value is `0`, which indicates an unset value. But it might already be non-zero. So instead, shouldn't this now be:
```go
if config.Optimism != nil && (config.Optimism.EIP1559DenominatorCanyon == nil || *config.Optimism.EIP1559DenominatorCanyon == 0) {
```
?

Also wondering if there's an existing test where we could add the three cases `nil`, `0`, and _non-zero_ to?

Reply:
Yes that was changing the logic. I understand now the override should only take effect when the value is unset so will accept your suggestion. Will look for tests to modify and check for the three cases

Reply:
Tests added here https://github.com/ethereum-optimism/op-geth/pull/345/commits/682f7bbefe61efbc813e01cd7b64821cb61ac128